### ISSUE TITLE
# Database Schema:

### ISSUE THREAD

**Issue by TahjDev** [EthanLam17/artify#2]
- [x] Contains correct datatypes
- [x] Contains appropriate constraints/details
    - [x] primary key
    - [x] not null
    - [x] unique
    - [x] indexed
    - [x] foreign key
- [x] Contains bullet points after the table that state which foreign keys will reference to which table, or references to the associations which will be made
- [x] foreign key and table name are lowercased, snake_cased and back_ticked
- [x] Correctly formatted
    - [x] schema is written in a table format
    - [x] the table's name are lowercased, snake_cased and back_ticked
    - [x] the table header column names are bolded
    - [x] columns names are lowercased and snaked_cased and back_ticked

**Comment by TahjDev** [comment]
- Can you make your table names bigger
- should you not have a joins table for playlists as well as likes?
- I see that spotify user's have a username can you add that to your schema
-

**Comment by EthanLam17** [comment]
Hey Tahj, is it preferred to have join tables rather than having foreign keys to connect associated elements? I had incorporated foreign keys thinking that was enough but if you could clarify that for me then I will include the necessary join tables.

**Comment by TahjDev** [comment]
How are you going to connect songs to a playlist. I see in your schema that you have user_id on the playlist table but I don't see anyway to reference songs. And when you do reference songs you'll need a joins table case one single doesn't belong to a playlist, a song may have many playlists that it has been added to. And a playlist can have many songs. 
Also I don't see anything about likes? can you like things?

**Comment by EthanLam17** [comment]
I have added a joins table connecting songs to playlists so that a song can be in multiple playlists and a playlist can have multiple songs. I intended the liked songs function to just be another playlist. When a user "likes"  a song, it would add it to the user's default playlist: "Liked Songs". I was thinking of having it be like this because Liked Songs seemed to be very similar to a regular playlist.
